### ISSUE TITLE
Prow: deck shouldn't report error message when job history symlink doesn't exist

### ISSUE THREAD

**Issue by chaodaiG** [kubernetes/test-infra#17636]
**What happened**:
Deck logs errors when it couldn't read job history symlink for a job

**What you expected to happen**:
Deck should differentiate the reason why it failed reading job history symlink.
- If file not exist, just log as info/debug
- If any other reason, log as error

**How to reproduce it (as minimally and precisely as possible)**:
Go to https://prow.knative.dev/job-history/knative-prow/pr-logs/directory/pull-knative-client-integration-tests, and deck would log a bunch of errors as some of the symlink files don't exist

**Please provide links to example occurrences, if any**:
See https://github.com/knative/test-infra/issues/2089

**Anything else we need to know?**:

**Comment by alvaroaleman** [comment]
/reopen
My fixes do not seem to work

**Comment by chaodaiG** [comment]
> /reopen
> My fixes do not seem to work

Curious how you tested your changes

**Comment by alvaroaleman** [comment]
By rolling it out :)

**Comment by chaodaiG** [comment]
Could find the PR updating k8s prow, assume it was tested on another prow instance?

**Comment by alvaroaleman** [comment]
Yeah, the openshift one: https://github.com/openshift/release/pulls?q=is%3Apr+Update+prow+and+other+images+as+necessary

We usually bump it at least daily

**Comment by chaodaiG** [comment]
Glad to know. Seems like knative prow isn't the only instance suffering from this issue :)

**Comment by stevekuznetsov** [comment]
Deck log spam is a feature, not a bug!

**Comment by chaodaiG** [comment]
> Deck log spam is a feature, not a bug!

Agree that it's not a bug. It would be totally fine if it logs as `debug` or `info` instead of `error`, as this makes us almost have to ignore all errors from deck

**Comment by alvaroaleman** [comment]
He was trolling. It is a bug. Its not an error in decks code or configuration, jobs sometimes not getting to the point of uploading logs or failing doing so will always happen sometimes.
